src/raftstore/store/peer.rs

```diff
@@ -459,13 +462,40 @@ impl Peer {
             ready.hs.take();
         }
 
+        if let Some(ref soft_state) = ready.ss {
+            if soft_state.leader_id == raft::INVALID_ID {
+                if self.leader_missing_time.is_none() {
+                    self.leader_missing_time = Some(Instant::now())
+                }
+            } else if self.is_initialized() {
```

Comment:
Even the peer is uninitialized, if it receives a raft message from leader, I think we should clear leader_missing_time too.